New Feature: add Character class

korifey91 opened the pull request.

korifey91 committed:
New Feature: add Character class

3AHAT0P commented on src/classes/Character.js:
```diff
@@ -0,0 +1,105 @@
+      this._offscreenCanvas = new OffscreenCanvas(this.mainSprite.width, this.mainSprite.height);
+      
+      return this;
+    })();
```
Please use async create function (aka fabricMethod) instead of async constructor

korifey91 committed:
New Feature: add Character class
Dev: add static create method for Character.js
Dev: add validations for flipbooks
Dev: add static create method for Flipbook.js
Dev: add Sprite class

korifey91 committed:
New Feature: add Character class
Dev: add jsdoc description for Character.create method

korifey91 committed:
New Feature: add Character class
Dev: fix jsdoc description for Character.create method
Dev: add jsdoc description for Character constructor

korifey91 committed:
New Feature: add Character class
Dev: refactor Flipbook.js

korifey91 committed:
New Feature: add Character class
Dev: add start\stop methods for Flipbook.js

korifey91 committed:
New Feature: add Character class
Dev: fix stop method in Flipbook.js

korifey91 committed:
New Feature: add Character class
Dev: add position to Character.js

korifey91 committed:
New Feature: add Character class
Dev: add _offscreenCanvas

korifey91 committed:
New Feature: add Character class
Dev: add actionHandling(not finished yet) - Character.js
Dev: add _offscreenCanvas - Flipbook.js
Dev: add currentSprite getter - Sprite.js

korifey91 committed:
New Feature: add Character class
Dev: add speed for Character.js
Dev: add functionality to set sprites for different directions in Character.js
Dev: add render method in Character.js
Dev: add _render method in Flipbook.js

korifey91 committed:
New Feature: add Character class
Dev: bug fix
Dev: memory leaks fix

korifey91 committed:
New Feature: add Character class
Dev: fix mirroring. But it's hard code now. This settings should be passed through a some options. TBD

korifey91 committed:
New Feature: add Character class
Dev: refactor folders and file names

korifey91 committed:
New Feature: add Character class
Dev: add event handling to Flipbook.js
Dev: fix _setOnChangeJumpFrame functionality in Character.js

korifey91 committed:
New Feature: add Scene class

korifey91 committed:
New Feature: add Scene class
Dev: add keypress event handling for jump
Dev: add check position to Scene

korifey91 committed:
New Feature: add Scene class
Dev: fix all move and jump bugs

korifey91 committed:
New Feature: add Scene class
Dev: add check any collisions in move process of Character.js
Dev: add addObject for Scene
Dev: add collisions detection functionality in Scene

korifey91 committed:
New Feature: add Scene class
Dev: add method to render objects in Scene
Dev: fix _changePosition in Character.js
